src/ReportManager/SchemaFormContext/SchemaForm/index.js

```diff
@@ -0,0 +1,70 @@
+import React, { forwardRef, useState } from 'react';
+
+import SchemaFormContextProvider from './SchemaFormContext';
+import Section from './fields/Section';
+
+import styles from './styles.module.scss';
+import { FORM_FIELDS_TYPES } from '../../constants';
+import { TextFieldValidators } from './fields/Text';
+import { getSchemaFieldUIType, isFieldActive } from './SchemaFormContext/utils';
+
+const FormFieldValidators = {
+  [FORM_FIELDS_TYPES.TEXT]: TextFieldValidators
+};
+
+const SchemaForm = ({ schema, formData = {}, onFormSubmit, className, onChange }, ref) => {
+
+  const [formErrors, setFormErrors] = useState({});
+
+  const handleOnSubmit = (e) => {
+    e.preventDefault();
+
+    const formErr = { ...formErrors };
+
+    for (const [fieldName, fieldValue] of Object.entries(formData)) {
```

Comment:
nit: not really an issue... But why are we using old `for`s instead of iterator methods?

Reply:
Just for personal preference, I have no problem changing to iterator methods

Reply:
Actually, I remember I wanted to break the iteration when a error was found, but that happens only in the second level for loop, so not sure I the code would look good mixing both techniques